inject/android.js

```diff
@@ -16,10 +25,47 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    const messageSecret = processedConfig.messageSecret
+    // Receives messages from the platform
+    const messageCallback = processedConfig.messageCallback
+    // Sends messages to the platform
+    const messageInterface = processedConfig.messageInterface
+
+    const wrappedUpdate = ((providedSecret, ...args) => {
+        if (providedSecret === messageSecret) {
+            update(...args)
+        }
+    }).bind()
+
+    Object.defineProperty(window, messageCallback, {
+        value: wrappedUpdate
+    })
+
     init(processedConfig)
```

Comment:
Nit: It would be better to move the init() call just below the `sendMessageToAndroid` lines. E.g.

```javascript
const sendMessageToAndroid = window[messageInterface].process.bind(window[messageInterface])
delete window[messageInterface]

init(processedConfig)

window.addEventListener('sendMessageProxy' + messageSecret, event => {
```